sdk/identity/azure-identity/azure/identity/_credentials/user_password.py

```diff
@@ -37,50 +33,28 @@ class UsernamePasswordCredential(PublicClientCredential):
           defines authorities for other clouds.
     :keyword str tenant_id: tenant ID or a domain associated with a tenant. If not provided, defaults to the
           'organizations' tenant, which supports only Azure Active Directory work or school accounts.
+    :keyword bool enable_persistent_cache: if True, the credential will store tokens in a persistent cache shared by
+         other user credentials. Defaults to False.
+    :keyword bool allow_unencrypted_cache: if True, the credential will fall back to a plaintext cache on platforms
+          where encryption is unavailable. Default to False. Has no effect when `enable_persistent_cache` is False.
     """
 
     def __init__(self, client_id, username, password, **kwargs):
         # type: (str, str, str, Any) -> None
+
+        # The base class will accept an AuthenticationRecord, allowing this credential to authenticate silently the
+        # first time it's asked for a token. However, we want to ensure this first authentication is not silent, to
+        # validate the given password. This class therefore doesn't document the authentication_record argument, and we
+        # discard it here.
+        kwargs.pop("authentication_record", None)
```

Comment:
In old code, I see 

for account in accounts:
            result = app.acquire_token_silent(scopes, account=account)
            if result:
                break

It seems to me we allow first time silent authentication?

Reply:
It may appear so but that was actually impossible because the cache was in memory; the credential always started with an empty one.